v2/contribute/decisions/oauth/0010-m2m-admin-access.mdx

```diff
@@ -0,0 +1,46 @@
+---
+id: "0010"
+title: Recommend access tokens using custom scopes for M2M
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Recommend access tokens using custom scopes for M2M
+
+<DecisionHeader status="proposed" lastUpdate="2023-05-11" created="2023-05-11" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+We want to remove/simplify the M2M guide that uses the JWT recipe.
+
+## Considered Options
+
+* Provide separate API for admin keys
+* Recommend access tokens using custom scopes for M2M
+
+## Decision Outcome
+
+Chosen option: Recommend access tokens using custom scopes for M2M
+
+* Fits the client credential flow of OAuth2 well.
+* Doesn't require a separate/dedicated recipe for M2M.
+
+## Pros and Cons of the Options
```

Comment:
related to this, how will can client credential flow be used by our users to allow them to issue API keys to their customers so that their customers can query our user's backend using the API key?

Usually, what i have seen is that our customer would give a client id + secret to their customers, who would then use the client credential flow to get a new access token, which can then be used to query our customer's application backend. In this case, since our access tokens are JWTs, how can our customers revoke them?

Finally, can this client credentials flow be used to just give one single API key to customers and using just that, they can directly access our customer's application's backend apis?

Reply:
1. They create an access token using the client credential flow - this access token can be handled asn an opaque "API key"
2. We've discussed that flow as well
    - It will be available (this is about what we recommend)
    - We've discussed that issuing a long lived token can be easier to use from the POV of the customer (of our customer)
3. They can be (and in this case should be) handled as opaque tokens. We expose a token validation endpoint for this purpose.
4. Yes - basically the issued access token is the api key.

Reply:
Understood.